x-pack/plugins/profiling/server/lib/setup/steps/get_cluster_settings_step.ts

```diff
@@ -17,14 +17,25 @@ export function getClusterSettingsStep({
     hasCompleted: async () => {
       const settings = await client.getEsClient().cluster.getSettings({});
 
-      return settings.persistent.search?.max_buckets === MAX_BUCKETS.toString();
+      const isProfilingTemplatesEnabled =
+        settings.persistent.xpack?.profiling?.templates?.enabled ?? false;
```

Comment:
This check can likely be replaced by querying the [new status API](https://github.com/elastic/elasticsearch/pull/96272) after it is merged.